python/src/trezorlib/cli/firmware.py

```diff
@@ -15,25 +15,49 @@
 # If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
 
 import sys
+import os
+from urllib.parse import urlparse
 
 import click
 import requests
 
+from typing import BinaryIO
+
 from .. import exceptions, firmware
-from . import with_client
+
+from ..client import TrezorClient
+from . import with_client, TrezorConnection
 
 ALLOWED_FIRMWARE_FORMATS = {
     1: (firmware.FirmwareFormat.TREZOR_ONE, firmware.FirmwareFormat.TREZOR_ONE_V2),
     2: (firmware.FirmwareFormat.TREZOR_T,),
 }
 
 
-def _print_version(version):
+def _print_version(version: dict) -> None:
     vstr = "Firmware version {major}.{minor}.{patch} build {build}".format(**version)
     click.echo(vstr)
 
 
-def validate_firmware(version, fw, expected_fingerprint=None):
+def _is_bootloader_onev2(client: TrezorClient) -> bool:
+    """Check if bootloader is capable of installing the Trezor One v2 firmware directly."""
+    f = client.features
+    bootloader_version = (f.major_version, f.minor_version, f.patch_version)
```

Comment:
This is no longer necessarily _bootloader_ version too.

AFAICT it doesn't affect functionality (firmware version 1.8.0 also must have bootloader 1.8.0 or better), but please rename the variable and modify docstring so that this is documented.

Reply:
Is it enough to rename the word "bootloader" with "device" both in the variable name and in docstring?

Reply:
"This is the case from bootloader version 1.8.0, and also holds for _firmware_ version 1.8.0 because that installs the appropriate bootloader."